proxy_connect.go

```diff
@@ -74,27 +74,24 @@ func (ic *connectInterceptor) connect(ctx context.Context, w http.ResponseWriter
 		}
 	}()
 
-	// Note - for CONNECT requests, we use the Host from the request URL, not the
-	// Host header. See discussion here:
-	// https://ask.wireshark.org/questions/22988/http-host-header-with-and-without-port-number
-	upstream, err = ic.dial(ctx, "tcp", req.URL.Host)
-	if err != nil {
-		fullErr := errors.New("Unable to dial upstream: %s", err)
-		log.Debug(fullErr)
-		respondBadGateway(w, fullErr)
-		return fullErr
-	}
-	closeUpstream = true
-
 	// Hijack underlying connection.
 	downstream, _, err = w.(http.Hijacker).Hijack()
 	if err != nil {
+		// If there's an error hijacking, that's almost certainly because the
+		// connection has died, so there's no point in trying to send some sort
+		// of error response.
 		fullErr := errors.New("Unable to hijack connection: %s", err)
```

Comment:
Actually, in the standard Go implementation of ResponseWriter, the only reason that a connection cannot be hijacked is that it has already been hijacked. If that happens, this indicates a programming error.